R/filter_panel_api.R

```diff
@@ -23,8 +23,12 @@
 #' @seealso [`teal_slice`]
 #'
 #' @examples
-#' library(MultiAssayExperiment)
-#' data(miniACC)
+#' if (requireNamespace("MultiAssayExperiment", quietly = TRUE)) {
+#'   library(MultiAssayExperiment)
+#'   data(miniACC)
+#' } else {
+#'   miniACC <- data.frame()
+#' }
```

Comment:
While I don't see a better way to handle this, I don't like it. Sure, the examples will work properly, but we pass a `data.frame` instead of a `MultiAssayExperiment`, that's quite a difference.

How about we limit this example to `data.frames`, maybe add `mtcars` for the sake of having two datasets? The downside then becomes we don't have a public-facing example involving an MAE. Perhaps move the filter specification for `miniACC` to the `teal_slice` docs?

Reply:
The implementation seemed like the lesser of evils. Keeping it as it is makes this example quite complex, but compact.

I do like your suggestion, we can also add `mtcars` and a new section just for MAE in the example. Keeping in line the "optional" aspect of this requirement, as well as prepare this to a possible [future break](https://github.com/insightsengineering/teal.slice/issues/500)